### PR TITLE
make compatible with active learning

### PR DESCRIPTION
**Pull request by nikita-0209**

### CODE REVIEW

File: gflownet/gflownet.py [hunk 1]
```diff
@@ -88,7 +88,9 @@ def __init__(
         self.logger = logger
         self.oracle_n = oracle.n
         # Buffers
-        self.buffer = Buffer(**buffer, env=self.env, make_train_test=not sample_only)
+        self.buffer = Buffer(
+            **buffer, env=self.env, make_train_test=not sample_only, logger=logger
```
Comment (nikita-0209) [inline gflownet/gflownet.py:92]: hack

File: gflownet/gflownet.py [hunk 2]
```diff
@@ -995,23 +998,25 @@ def uniform_distribution(self, states):
 
 
 def batch2dict(batch, env, get_uncertainties=False, query_function="Both"):
-    batch = np.asarray(env.state2oracle(batch))
+    batch = [env.state2proxy(state) for state in batch]
```
Comment (nikita-0209) [inline gflownet/gflownet.py:1001]: Quick Fix

File: config/main.yaml [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-  - env: amp
+  - env: grid
   - gflownet: flowmatch
-  - proxy: amp
+  - proxy: grid_corners
   - logger: wandb
-  - user: nikita
-
+  
+user: nikita
```
Comment (alexhernandezgarcia) [inline config/main.yaml:8]: move to defaults, and have default=alex :)

File: gflownet/envs/grid.py [hunk 4]
```diff
@@ -74,6 +74,26 @@ def __init__(
...
+            self.reward_norm,
+            self.reward_norm_std_mult,
+            self.reward_func,
+            self.denorm_proxy,
+            self.energies_stats,
+            self.proxy,
+            self.oracle,
+            self.proxy_state_format,
+        )
+
```
Comment (alexhernandezgarcia) [inline gflownet/envs/grid.py:96]: 1. Make sure we need custom copy(). Ideally no
2. If so, implement in base.py

File: gflownet/gflownet.py [hunk 5]
```diff
@@ -684,7 +683,7 @@ def train(self):
         loss_term_ema = None
         loss_flow_ema = None
         # Generate list of environments
-        envs = [copy.deepcopy(self.env).reset() for _ in range(self.batch_size)]
+        envs = [self.env.copy() for _ in range(self.batch_size)]
```
Comment (alexhernandezgarcia) [inline gflownet/gflownet.py:686]: Update if custom copy not needed.

File: gflownet/gflownet.py [hunk 6]
```diff
@@ -843,6 +842,7 @@ def log_iter(
         oracle_batch, oracle_times = self.sample_batch(
             self.env, self.oracle_n, train=False
         )
+
```
Comment (alexhernandezgarcia) [inline gflownet/gflownet.py:845]: do we need this line?

File: gflownet/utils/logger.py [hunk 7]
```diff
@@ -163,7 +161,7 @@ def log_sampler_train(
             self.log_metrics(
                 train_metrics,
                 use_context=use_context,
-                step=step,
+                # step=step,
```
Comment (alexhernandezgarcia) [inline gflownet/utils/logger.py:164]: ugly

File: gflownet/utils/logger.py [hunk 8]
```diff
@@ -187,7 +185,7 @@ def log_sampler_test(
             self.log_metrics(
                 test_metrics,
                 use_context=use_context,
-                step=step,
+                # step=step,
```
Comment (alexhernandezgarcia) [inline gflownet/utils/logger.py:188]: remove

File: gflownet/utils/logger.py [hunk 9]
```diff
@@ -221,7 +219,7 @@ def log_sampler_loss(
         self.log_metrics(
             loss_metrics,
             use_context=use_context,
-            step=step,
+            # step=step,
```
Comment (alexhernandezgarcia) [inline gflownet/utils/logger.py:222]: remove

**Pull request merged**